in_toto/runlib.go

```diff
@@ -48,6 +49,35 @@ func RecordArtifact(path string) (map[string]interface{}, error) {
 	return hashedContentsMap, nil
 }
 
+// ApplyExcludePatterns takes paths and excludePatterns as input
+// and compares both. If the excludePattern matches a file path
+// the file will not be included.
+// This behavior is similar to git's famous gitignore file.
+func ApplyExcludePatterns(paths []string, excludePatterns []string) ([]string, error) {
+	if excludePatterns == nil {
+		return paths, nil
+	}
+	// parses the exclude patterns into pattern type object
+	var ps []Pattern
+	var included []string
+	for _, element := range excludePatterns {
+		temp := ParsePattern(element, nil)
+		ps = append(ps, temp)
+	}
+	m := NewMatcher(ps)
+	for _, path := range paths {
+		fileInfo, err := os.Stat(path)
+		if err != nil {
+			return nil, err
+		}
+		match := m.Match([]string{path}, fileInfo.IsDir())
```

Comment:
Why do we care if the path is a dir?

Reply:
This is just how the gitignore matcher works: https://pkg.go.dev/gopkg.in/src-d/go-git.v4/plumbing/format/gitignore?tab=doc#Matcher

With a normal regexp matcher we would not need `fileInfo.IsDir()`: https://pkg.go.dev/gopkg.in/src-d/go-git.v4/plumbing/format/gitignore?tab=doc#Matcher

I guess the former dev chose the gitignore matcher, because it implements most of the filesystem path matching already (different platforms: Linux, Windows, Mac, easier path matching, etc)